File: Remove Nth Node From End of List.cpp
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* fast = head;
        ListNode* slow = head;
        while(n--){
            fast = fast->next;
        }
        if(fast==NULL){
            return slow->next;
        }else{
            while(fast->next){
                slow = slow->next;
                fast = fast->next;
            }
            slow->next = slow->next->next;
            return head;
        }
    }
};

File: Remove Nth Node From End of List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        if head.next == None:
            return None
        size = 0
        curr = head
        while curr:
            size += 1
            curr = curr.next

        if size-n == 0:
            head = head.next
            
        temp = head
        for index in range(size-n-1):
            temp = temp.next
        
        if temp and temp.next:
            temp.next = temp.next.next
        return head
